builtin/providers/ultradns/resource_ultradns_record.go

```diff
@@ -0,0 +1,181 @@
+package ultradns
+
+import (
+	"fmt"
+	"github.com/Ensighten/udnssdk"
+	"github.com/hashicorp/terraform/helper/schema"
+	"log"
+	"strconv"
+)
+
+func resourceUltraDNSRecord() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceUltraDNSRecordCreate,
+		Read:   resourceUltraDNSRecordRead,
+		Update: resourceUltraDNSRecordUpdate,
+		Delete: resourceUltraDNSRecordDelete,
+
+		Schema: map[string]*schema.Schema{
+			"zone": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"hostname": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"type": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"rdata": &schema.Schema{
+				Type:     schema.TypeList,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+
+			"ttl": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "3600",
+			},
+		},
+	}
+}
+
+func resourceUltraDNSRecordCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*udnssdk.Client)
+	newRecord := &udnssdk.RRSet{
+		OwnerName: d.Get("name").(string),
+		RRType:    d.Get("type").(string),
+	}
+	rdata := d.Get("rdata").([]interface{})
+	rdatas := make([]string, len(rdata))
+	for i, j := range rdata {
+		rdatas[i] = j.(string)
+	}
+	newRecord.RData = rdatas
+	ttl := d.Get("ttl").(string)
+	newRecord.TTL, _ = strconv.Atoi(ttl)
+
+	log.Printf("[DEBUG] UltraDNS RRSet create configuration: %#v", newRecord)
+
+	_, err := client.RRSets.CreateRRSet(d.Get("zone").(string), *newRecord)
+	recId := fmt.Sprintf("%s.%s", d.Get("name").(string), d.Get("zone").(string))
+	if err != nil {
+		return fmt.Errorf("Failed to create UltraDNS RRSet: %s", err)
+	}
```

Comment:
Does a create take place immediately? Is there any API we need to poll to ensure the recordset is fully created?